automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/SSH.scala

```diff
@@ -46,38 +55,99 @@ object SSH {
     Resource.make(makeTunnel)(tunnel => loggerIO.info("Closing tunnel") >> closeTunnel(tunnel))
   }
 
-  final case class SessionAndClient(makeSession: Session, client: SSHClient)
+  final case class SSHSession(session: Session, client: SSHClient)
   // Note that a session is a one time use resource, and only supports one command execution
-  def makeSSHSession(hostName: String, port: Int): Resource[IO, SessionAndClient] = {
+  private def startSSHConnection(hostName: String, port: Int, sshConfig: SSHRuntimeInfo): Resource[IO, SSHSession] = {
     val sessionAndClient = for {
-      _ <- loggerIO.info(
-        s"Making ssh client u ${LeonardoConfig.Leonardo.vmUser} p ${LeonardoConfig.Leonardo.vmPassword}"
-      )
       client <- IO(new SSHClient)
       _ <- loggerIO.info(s"Adding host key verifier for shh client}")
       _ <- IO(client.addHostKeyVerifier(new PromiscuousVerifier()))
-      _ <- loggerIO.info("Connecting via ssh client")
+
+      _ <- loggerIO.info(s"Connecting via ssh client hostname ${hostName} port $port")
       _ <- IO(client.connect(hostName, port))
-      _ <- loggerIO.info("Authenticating ssh client via password")
-      _ <- IO(client.authPassword(LeonardoConfig.Leonardo.vmUser, LeonardoConfig.Leonardo.vmPassword))
+
+      _ <- loggerIO.info("Authenticating ssh client ")
+      _ <-
+        if (sshConfig.cloudProvider == CloudProvider.Azure)
```

Comment:
Yeah, the subsequent else block handles this flow for google

Reply:
Probably NOT for this code review (aka not a refactor code-review) ...
but this is a code smell - if-else blocks based on things that could/should otherwise be handled by an object of a certain type.

... and yes, I know we have this style of code everywhere

Reply:
Yeah, that's true, we do need to be better about higher levels of abstraction